### PR TITLE
docs: fix typos

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

**Comment by adrian5** [comment]
Contributing simple doc PRs has become too much of a technical headache so I'll comment here: There's a stray bullet in [lsp.txt](https://github.com/neovim/neovim/blob/master/runtime/doc/lsp.txt#L1121).

Formatting is still all over the place, e.g. why aren't headings equally right-aligned ([1](https://github.com/neovim/neovim/blob/master/runtime/doc/lsp.txt#L22), [2](https://github.com/neovim/neovim/blob/master/runtime/doc/lsp.txt#L447))? Capitalization and colons are used arbitrarily too. When I open the nvim help this improvised style always jumps out at me, but I guess I'm in the minority.

**Comment by dundargoc** [comment]
> Contributing simple doc PRs has become too much of a technical headache

@adrian5 can you elaborate? Are you referring to setting up documentation generation with `gen_vimdoc.py` or something else?

> Formatting is still all over the place, e.g. why aren't headings equally right-aligned ([1](https://github.com/neovim/neovim/blob/master/runtime/doc/lsp.txt#L22), [2](https://github.com/neovim/neovim/blob/master/runtime/doc/lsp.txt#L447))?

Because we need to improve our documentation consistency and we need to automate it, which takes time and effort.

**Comment by adrian5** [comment]
> Can you elaborate? Are you referring to setting up documentation generation with gen_vimdoc.py or something else?

Right, that part. Though I saw that you're absorbing small PRs into drafts like these, so if I can get away without the docgen part it would be simple enough again.

> Because we need to improve our documentation consistency and we need to automate it, which takes time and effort.

Is there some kind of guideline on that yet on what's being aimed for?

**Comment by clason** [comment]
> Formatting is still all over the place, e.g. why aren't headings equally right-aligned ([1](https://github.com/neovim/neovim/blob/master/runtime/doc/lsp.txt#L22), [2](https://github.com/neovim/neovim/blob/master/runtime/doc/lsp.txt#L447))?

Because these files are automatically generated, and some layout is just very hard to parse. Improvements very welcome.

>  Capitalization and colons are used arbitrarily too. When I open the nvim help this improvised style always jumps out at me, but I guess I'm in the minority.

Again, this is a community effort, with a lot of contributors over many years. And some files are taken from Vim, where Bram has his own style (that we deliberately leave untouched). Going through the ~120 help files and "fixing" all these minor style issues is a lot of work (and noise), so it's not worth it just for visual nicety.

**Comment by adrian5** [comment]
Hmm I see, I think that answers my question.

**Comment by adrian5** [comment]
> But I don't see what's so hard about running ./scripts/gen_vimdoc.py from the repo root?

Apparently I need msgpack (pip), doxygen (pacman) and nvim-0.9 (which is where I abandoned the effort).
